service/controller/clusterapi/v17/resources/awsclusterconfig/create.go

```diff
@@ -135,12 +160,33 @@ func (r *Resource) mapClusterToAWSClusterConfig(awsClusterConfig corev1alpha1.AW
 		},
 	}
 
-	// TODO: Workers shall be added when we have better understanding towards template structure.
+	var workers []corev1alpha1.AWSClusterConfigSpecGuestWorker
+	for _, md := range machineDeployments {
+		if md.Spec.Replicas == nil {
+			continue
+		}
+
+		for i := 0; i < int(*md.Spec.Replicas); i++ {
+			w := corev1alpha1.AWSClusterConfigSpecGuestWorker{
+				AWSClusterConfigSpecGuestNode: corev1alpha1.AWSClusterConfigSpecGuestNode{
+					ID:           fmt.Sprintf("%s-%d", key.MachineDeployment(&md), i),
```

Comment:
I'm not really sure if this is used for anything, but node pool ID + running number felt ~right.

Reply:
Well, what is this ID even? What is this used for?

Reply:
The ID is from the old times when `api` generated an id for each worker and this was somehow tracked along the way to the provider specific config as well. As said, I'm not sure if it's used for anything because `cluster-operator` doesn't create those provider configs.

Reply:
Is the ID actually helpful in any scenario?
Maybe it is nice to reference different MDs when debugging - other than that I don't really see a use for it.

Reply:
I feel like it might not be used for anything anymore on AWS. Then I would deprecate it and not fill it for the Cluster API types.

Reply:
This ID is not a `MachineDeployment` but an ID for individual worker. As we are unsure if this has of any use and on the other hand, when the cluster itself came up successfully already without any workers (besides the managed apps that do need these), I'll remove this. Let's see how it goes then. :wink: